projects/components/src/common/activity-reporter/activity-reporter.spec.ts

```diff
@@ -0,0 +1,169 @@
+/*!
+ * Copyright 2020 VMware, Inc.
+ * SPDX-License-Identifier: BSD-2-Clause
+ */
+
+import { ActivityReporter, ActivityStatus, ActivityResponse } from './activity-reporter';
+
+let resolve;
+let reject;
+let promise;
+
+let reporter: ActivityReporter;
+
+let start;
+let reportError;
+let reportSuccess;
+
+describe('ActivityReporter', () => {
+    beforeEach(() => {
+        promise = new Promise((promiseResolve, promiseReject) => {
+            resolve = stuff => promiseResolve(stuff);
+            reject = stuff => promiseReject(stuff);
+        });
+
+        reporter = new SimpleReporter();
+        start = spyOn(reporter, 'startActivity');
+        reportError = spyOn(reporter, 'reportError');
+        reportSuccess = spyOn(reporter, 'reportSuccess');
+    });
+
+    class SimpleReporter extends ActivityReporter {
+        reportError(errorText: string): void {}
+
+        reportSuccess(successMessage?: string): void {}
+
+        startActivity(): void {}
+    }
+
+    describe('monitorActivity', () => {
+        it('can monitor an activity with success', done => {
+            const getPromise = reporter.monitorActivity(promise);
+            expect(start).toHaveBeenCalledTimes(1);
+
+            getPromise.then(result => {
+                expect(result.status).toEqual(ActivityStatus.READY);
+                expect(reportSuccess).toHaveBeenCalledTimes(1);
+                done();
+            });
+
+            resolve({
+                status: ActivityStatus.READY,
+            });
+        });
+
+        it('can monitor an activity resolved failure', done => {
```

Comment:
Couldn't we use async await? Or at least return a Promise? done was created before we had Promises and async/await.